### PR TITLE
Service: Power supply

### PR DESCRIPTION
**Pull request by jamesadevine**
This pull request defines a new service.

### CODE REVIEW

File: services/powersupply.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Power supply
+    
+    identifier: 0x1f40375f
+    
+A power supply with a static or variable voltage range
+
+## Registers
+
+    rw enabled: u8 @ 0x81;
```
Comment (pelikhan) [inline services/powersupply.md:9]: this should be the intensity register

File: services/powersupply.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+Turns the power supply on with a non-zero value, off with a zero value. 
+    
+    rw output_voltage: u16.16 @0x82
```
Comment (pelikhan) [inline services/powersupply.md:13]: probably use the value register here as well

File: services/powersupply.md (hunk 1 above)
Comment (pelikhan) [reply services/powersupply.md:9]: boolean?

**Comment by jamesadevine** [comment]
am I now misusing: min_reading and max_reading?

**Comment by jamesadevine** [comment]
> am I now misusing: min_reading and max_reading?

A: yes.

**Comment by pelikhan** [comment]
yup min_value, max_value best

**Comment by jamesadevine** [comment]
Maybe. Happy to stick to just output voltage for now?

James
________________________________

**Comment by pelikhan** [comment]
output voltage yes

**Comment by pelikhan** [comment]
i16.16?
________________________________

**Comment by mmoskal** [comment]
I would go `i22.10 mV` - with `mV` you don't need `.16` but if it's a generic power supply you don't want to limit to 65V

**Comment by pelikhan** [comment]
Run "yarn buildspecs" from jacdac-docs

**Pull request merged**